plugins/woocommerce/tests/performance/tests/gh-action-pr-requests.js

```diff
@@ -248,6 +256,118 @@ export const options = {
 	},
 };
 
+export function setup() {
+	/**
+	 * Convert Cart & Checkout pages to shortcode
+	 */
+	function useCartCheckoutShortcodes() {
+		/**
+		 * A WordPress page.
+		 * @typedef {Object} WPPage
+		 * @property {number} id
+		 * @property {string} slug
+		 */
+
+		let defaultHeaders;
+
+		/**
+		 * @type {WPPage[]}
+		 */
+		let wp_pages;
+
+		/**
+		 * @type {WPPage}
+		 */
+		let cart;
+
+		/**
+		 * @type {WPPage}
+		 */
+		let checkout;
+
+		function setDefaultHeaders() {
+			const credentials = `${ admin_username }:${ admin_password }`;
+			const encodedCredentials = encoding.b64encode( credentials );
+			defaultHeaders = {
+				Authorization: `Basic ${ encodedCredentials }`,
+				'Content-Type': 'application/json',
+			};
+		}
+
+		function listWPPages() {
+			const url = `${ base_url }/wp-json/wp/v2/pages`;
+			const body = {
+				_fields: [ 'id', 'slug' ],
+				context: 'edit',
+			};
+			const params = {
+				headers: defaultHeaders,
+			};
+			const response = http.get( url, body, params );
+			check( response, {
+				'WP pages list obtained': ( r ) => r.status === 200,
+			} );
+			wp_pages = response.json();
+		}
+
+		function findCartCheckoutPage() {
+			cart = wp_pages.find( ( page ) => page.slug === 'cart' );
+			checkout = wp_pages.find( ( page ) => page.slug === 'checkout' );
+		}
+
+		function convertToCartCheckoutShortcode() {
+			const url_cart = `${ base_url }/wp-json/wp/v2/pages/${ cart.id }`;
+			const url_checkout = `${ base_url }/wp-json/wp/v2/pages/${ checkout.id }`;
+
+			const body_cart = {
+				content: {
+					raw: '<!-- wp:shortcode -->[woocommerce_cart]<!-- /wp:shortcode -->',
+				},
+			};
+			const body_checkout = {
+				content: {
+					raw: '<!-- wp:shortcode -->[woocommerce_checkout]<!-- /wp:shortcode -->',
+				},
+			};
+
+			const body_cart_str = JSON.stringify( body_cart );
+			const body_checkout_str = JSON.stringify( body_checkout );
+
+			const params = {
+				headers: defaultHeaders,
+			};
+
+			const response_cart = http.post( url_cart, body_cart_str, params );
+			const response_checkout = http.post(
+				url_checkout,
+				body_checkout_str,
+				params
+			);
+
+			check( response_cart, {
+				'cart shortcode set': ( r ) =>
+					r.status >= 200 && r.status < 300,
+			} );
+			check( response_checkout, {
+				'checkout shortcode set': ( r ) =>
+					r.status >= 200 && r.status < 300,
+			} );
+		}
+
+		setDefaultHeaders();
+
+		listWPPages();
+
+		findCartCheckoutPage();
+
+		convertToCartCheckoutShortcode();
+	}
+
+	useCartCheckoutShortcodes();
```

Comment:
I don't have a strong background working with our performance tests in WooCommerce, but looking at how other tests are designed, it seems that these functions should be defined in a separate place within `performance/requests/merchant` or `performance/requests/shopper` (probably merchant in this case) and then those functions should be used in this `gh-action-pr-requests.js` file.

We might want to keep this code consistent, so I'd suggest following the approach mentioned above.

Reply:
Yup, I agree with having a separate js file for this. I'm thinking of maybe having a new `setup` folder inside `plugins/woocommerce/tests/performance` that will house setup files like this. Will make this change too.

Reply:
Done. I created a separate `plugins/woocommerce/tests/performance/setup/cart-checkout-shortcode.js` file for this. Thanks :)